### PR TITLE
memory/patcher: Added missed checking for returning value of SYS_brk calling

### PR DESCRIPTION
**Pull request by nikitaxgusev**
Signed-off-by: Nikita Gusev <nikita.gusev@intel.com>

### CODE REVIEW

**Comment by hoopoepg** [comment]
ok to test

File: opal/mca/memory/patcher/memory_patcher_component.c [hunk 1]
```diff
@@ -591,6 +591,9 @@ static int patcher_open (void)
 
 #if defined (SYS_brk)
     rc = opal_patcher->patch_symbol ("brk", (uintptr_t)intercept_brk, (uintptr_t *) &original_brk);
+    if (OPAL_SUCCESS != rc) {
```
Comment (hoopoepg) [inline opal/mca/memory/patcher/memory_patcher_component.c]: rc will be returned in any case, here evaluation of returned value is not needed

Comment (nikitaxgusev) [reply opal/mca/memory/patcher/memory_patcher_component.c]: @hoopoepg ok. I would like to support the code style. Should I close it?

Comment (hoopoepg) [reply opal/mca/memory/patcher/memory_patcher_component.c]: let's wait for another opinions :)

Comment (hjelmn) [reply opal/mca/memory/patcher/memory_patcher_component.c]: The code was intentionally written this way. The return code is already passed up.

**Pull request closed without merging**